### PR TITLE
Added instructions to run phpunit tests within CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by Lyndonbazza**

### CODE REVIEW

**Comment by Lyndonbazza** [comment]
I have run make docs and it does not work - I am assuming I'm doing something wrong again...

Please supply instructions on this.

File: project/CONTRIBUTING.md [hunk 1]
```diff
@@ -97,6 +97,19 @@ Some rules have to be respected about the test:
 * Most of the time, the test class should have the same name as the targeted class, suffixed by `Test`.
 * The `@expectedException*` annotations are prohibited. Use `PHPUnit_Framework_TestCase::setExpectedException()`.
 
+Setup phpunit to run the tests using:
+```
+composer install
+wget "https://phar.phpunit.de/phpunit.phar" --output-document="/usr/local/bin/phpunit"
+chmod ugo+x "/usr/local/bin/phpunit"
+```
+You may need to use sudo or choose a location to install phpunit to and add to your $PATH
```
Comment (greg0ire) [inline project/CONTRIBUTING.md]: Missing dot at the end of the sentence. Also, add "it" to your $PATH, maybe?

**Comment by greg0ire** [comment]
When asked for changes don't close the PR and reopen one. You can just push new commits to the branch you created.

**Comment by greg0ire** [comment]
> I have run make docs and it does not work - I am assuming I'm doing something wrong again...

Oh about the docs, don't worry, [I have it covered](https://github.com/sonata-project/dev-kit/pull/203).

**Comment by greg0ire** [comment]
If you feel up to it, squash your commits together, otherwise we will do it for you. Instructions on how to do that are on the very file you edited ;)

**Comment by Lyndonbazza** [comment]
Let me do this - I'll figure it out

**Comment by core23** [comment]
Not sure if we need this in our docs. For me this are the basics of PHP development.

**Comment by Lyndonbazza** [comment]
I ran the tests incorrectly using a simple unit test script that was downloaded as a dependency.

This made it easier for me to understand how to contribute to the project.

It's basic, but helpful.

**Comment by greg0ire** [comment]
@core23 : see https://github.com/sonata-project/SonataAdminBundle/pull/4237

**Comment by sstok** [comment]
FYI. phpunit-simple is for Symfony only (without Prophecy and YAML), this solves the circular dependency of phpunit depending on Symfony YAML and testing Symfony YAML, will use the phpunit's bundled version.

Instead of asking someone to copy/paste a number of commands, you can also provide a shell script. And I highly recommend to verify the download! https://github.com/defuse/php-encryption/blob/master/test/phpunit.sh

**Comment by Lyndonbazza** [comment]
I'm happy to contribute further here.

If I should create a shell script, please let me know. I am planning to try to get in to contributing more here so the fewer barriers to this, the better.

Thoughts?

**Comment by greg0ire** [comment]
It would be great to resolve #197 first, and it might be best to use phpunit-simple : 

> It automatically selects the best version to run: PHPUnit 4.x when run with PHP <=5.5 and PHPUnit 5.x when run with PHP >=5.6;

Which would be one less thing to maintain for us

**Comment by greg0ire** [comment]
@Lyndonbazza : I wouldn't recommend making a script : this is a one-time operation for developers, no need to automate it, so let's not complicate it. Instead, I would recommend [linking to this document](https://phpunit.de/manual/current/en/installation.html#installation.phar.verification) as an optional step.

**Comment by Lyndonbazza** [comment]
I will ammend the PR and add the optional link to verify the download.

In terms of #197 I think this is a small interim fix until you've made that decision.

If that's not what's wanted, let me know. Otherwise I will try to sort this today.

**Comment by greg0ire** [comment]
I think you're right, #197 has been open for ages, and has not been acted upon. Now that there are several ways to run phpunit, clarifying what is the current way to go might be useful. What do you think @core23?

File: project/CONTRIBUTING.md [hunk 2]
```diff
@@ -97,6 +97,21 @@ Some rules have to be respected about the test:
 * Most of the time, the test class should have the same name as the targeted class, suffixed by `Test`.
 * The `@expectedException*` annotations are prohibited. Use `PHPUnit_Framework_TestCase::setExpectedException()`.
 
+Setup phpunit to run the tests using:
+```
+composer install
+wget "https://phar.phpunit.de/phpunit.phar" --output-document="/usr/local/bin/phpunit"
```
Comment (Soullivaneuh) [inline project/CONTRIBUTING.md]: This should not be here, agree with @core23.

If you really want to specify it, it should be a indication of global phar usage with a link to official instructions.

**Comment by Soullivaneuh** [comment]
I think all the description is not necessary.

If you really want to, you can add [here](https://github.com/sonata-project/dev-kit/pull/220/files#diff-55aa24e78df8010aea154e9506cab16dR80) something like this:

> Unit tests are run with [PHPUnit](https://phpunit.de/).

Then, the contributor clicks on the link and follows official instructions.

**Comment by sstok** [comment]
I minor update on my comment, you can use phpunit-simple just make sure to not remove prophecy.
http://symfony.com/blog/how-to-solve-phpunit-issues-in-symfony-3-2-applications

```bash
$ export SYMFONY_PHPUNIT_REMOVE="symfony/yaml"
$ vendor/bin/simple-phpunit
```

File: project/CONTRIBUTING.md [hunk 3]
```diff
@@ -97,6 +97,21 @@ Some rules have to be respected about the test:
 * Most of the time, the test class should have the same name as the targeted class, suffixed by `Test`.
 * The `@expectedException*` annotations are prohibited. Use `PHPUnit_Framework_TestCase::setExpectedException()`.
 
+Setup phpunit to run the tests using:
+```
```
Comment (Soullivaneuh) [inline project/CONTRIBUTING.md]: Missing a blank line before this tag.

File: project/CONTRIBUTING.md [hunk 4]
```diff
@@ -97,6 +97,21 @@ Some rules have to be respected about the test:
...
+composer install
+wget "https://phar.phpunit.de/phpunit.phar" --output-document="/usr/local/bin/phpunit"
+chmod ugo+x "/usr/local/bin/phpunit"
+```
+You may need to use sudo or choose a location to install phpunit to and add it to your $PATH.
+
+Note, you should verfiy your download using the instructions at [phpunit.de](https://phpunit.de/manual/current/en/installation.html#installation.phar.verification).
+
+To run all tests, simply run:
+```
```
Comment (Soullivaneuh) [inline project/CONTRIBUTING.md]: Also missing blank line before this tag.

File: project/CONTRIBUTING.md [hunk 5]
```diff
@@ -97,6 +97,21 @@ Some rules have to be respected about the test:
 * Most of the time, the test class should have the same name as the targeted class, suffixed by `Test`.
 * The `@expectedException*` annotations are prohibited. Use `PHPUnit_Framework_TestCase::setExpectedException()`.
 
+Setup phpunit to run the tests using:
+```
+composer install
+wget "https://phar.phpunit.de/phpunit.phar" --output-document="/usr/local/bin/phpunit"
+chmod ugo+x "/usr/local/bin/phpunit"
+```
```
Comment (Soullivaneuh) [inline project/CONTRIBUTING.md]: Missing blank line after this tag.

**Comment by Soullivaneuh** [comment]
I did a review, but stick to this comment anyway: https://github.com/sonata-project/dev-kit/pull/220#issuecomment-272712394

**Comment by core23** [comment]
Can we close this?

**Comment by greg0ire** [comment]
Feel free to

**Pull request closed without merging**